geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy

```diff
@@ -2119,12 +2127,126 @@ IProcess rasterizeIndicators() {
+                    } else {
+                        info "Cannot compute the roof fraction distribution."
+                    }
+                }
+
+                if(list_indicators*.toUpperCase().contains("FRONTAL_AREA_INDEX") && buildingTable){
+                    if(!createScalesRelationsGridBl){
+                        // Create the relations between grid cells and buildings
+                        createScalesRelationsGridBl = Geoindicators.SpatialUnits.spatialJoin()
+                        if (!createScalesRelationsGridBl([datasource              : datasource,
+                                                          sourceTable             : buildingTable,
+                                                          targetTable             : gridTableName,
+                                                          idColumnTarget          : grid_column_identifier,
+                                                          prefixName              : prefixName,
+                                                          nbRelations             : null])) {
+                            info "Cannot compute the scales relations between buildings and grid cells."
+                            return
+                        }
+                    }
+                    def frontalAreaIndexDistribution = Geoindicators.RsuIndicators.frontalAreaIndexDistribution()
+                    if (frontalAreaIndexDistribution(
+                            [buildingTable               : createScalesRelationsGridBl.results.outputTableName,
+                             rsuTable                    : gridTableName,
+                             idRsu                       : grid_column_identifier,
+                             listLayersBottom            : [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50],
+                             numberOfDirection           : 12,
+                             prefixName                  : prefixName,
+                             datasource                  : datasource])) {
+                        indicatorTablesToJoin.put(frontalAreaIndexDistribution.results.outputTableName, grid_column_identifier)
+                    } else {
+                        info "Cannot compute the frontal area index."
+                    }
+                }
+
+                if(list_indicators*.toUpperCase().contains("SEA_LAND_FRACTION") && seaLandMaskTableName) {
+                    // If only one type of surface (land or sea) is in the zone, no need for computational fraction calculation
+                    def nbTypes = datasource.firstRow("""SELECT COUNT(DISTINCT($seaLandTypeField)) AS nb_types FROM $seaLandMaskTableName""").NB_TYPES
+                    if (nbTypes == 1) {
+                        def uniqueSurfaceType = datasource.firstRow("""SELECT DISTINCT($seaLandTypeField) AS unique_surface_type FROM $seaLandMaskTableName""").UNIQUE_SURFACE_TYPE
```

Comment:
You can reduce to one query with

rows = datasource.rows(SELECT COUNT(*) AS nb_types FROM $seaLandMaskTableName group by ($seaLandTypeField)"""))

and then use the rows to check if there is one seaLandTypeField equals to sea